code/modules/buildmode/submodes/advanced.dm

```diff
@@ -47,6 +47,10 @@
 			log_admin("Build Mode: [key_name(c)] modified [T] in [AREACOORD(object)] to [objholder]")
 			T = T.ChangeTurf(objholder)
 			T.setDir(BM.build_dir)
+		else if(ispath(objholder, /obj/effect/turf_decal))
+			var/turf/T = get_turf(object)
+			T.AddElement(/datum/element/decal, initial(objholder.icon), initial(objholder.icon_state), BM.build_dir, FALSE, initial(objholder.color), null, null, initial(objholder.alpha))
```

Comment:
can't you achieve this by spawning in the decal itself, and letting it do the work?